flink-runtime/src/test/java/org/apache/flink/runtime/leaderelection/DefaultLeaderElectionServiceTest.java

```diff
@@ -180,25 +186,117 @@ void testCloseGrantDeadlock() throws Exception {
     }
 
     @Test
-    void testGrantCallWhileInstantiatingDriver() throws Exception {
-        final UUID expectedLeaderSessionID = UUID.randomUUID();
+    void testLazyDriverInstantiation() throws Exception {
+        final AtomicBoolean driverCreated = new AtomicBoolean();
         try (final DefaultLeaderElectionService testInstance =
                 new DefaultLeaderElectionService(
                         (listener, errorHandler) -> {
-                            listener.isLeader(expectedLeaderSessionID);
+                            driverCreated.set(true);
                             return TestingLeaderElectionDriver.newNoOpBuilder()
                                     .build(listener, errorHandler);
                         },
                         fatalErrorHandlerExtension.getTestingFatalErrorHandler(),
                         Executors.newDirectExecutorService())) {
-            testInstance.startLeaderElectionBackend();
+            assertThat(driverCreated)
+                    .as("The driver shouldn't have been created during service creation.")
+                    .isFalse();
+
+            try (final LeaderElection leaderElection =
+                    testInstance.createLeaderElection("contender-id")) {
+                assertThat(driverCreated)
+                        .as(
+                                "The driver shouldn't have been created during LeaderElection creation.")
+                        .isFalse();
+
+                leaderElection.startLeaderElection(
+                        TestingGenericLeaderContender.newBuilder().build());
+                assertThat(driverCreated)
+                        .as(
+                                "The driver should have been created when registering the contender in the LeaderElection.")
+                        .isTrue();
+            }
+        }
+    }
+
+    @Test
+    void testReuseOfServiceIsRestricted() throws Exception {
+        final DefaultLeaderElectionService testInstance =
+                new DefaultLeaderElectionService(
+                        new TestingLeaderElectionDriver.Factory(
+                                TestingLeaderElectionDriver.newNoOpBuilder()));
+
+        // The driver hasn't started, yet, which prevents the service from going into running state.
+        // This results in the close method not having any effect.
+        testInstance.close();
+
+        try (final LeaderElection leaderElection =
+                testInstance.createLeaderElection("contender-id")) {
```

Comment:
Wondering whether this shouldn't fail with an exception. Whether a driver ended up being created or not shouldn't matter as to whether a closed LES can still be used.